src/filewatch/src/filewatcher.cpp

```diff
@@ -161,23 +161,24 @@ class EfswFileWatcher final : public efsw::FileWatchListener {
 
         const QFileInfo fileInfo = QFileInfo( fullFileName );
 
-        const auto filename = fileInfo.fileName().toStdString();
         const auto directory = fileInfo.absolutePath().toStdString();
 
         auto watchedDirectory
             = std::find_if( watchedPaths_.begin(), watchedPaths_.end(),
                             [ &directory ]( const auto& wd ) { return wd.name == directory; } );
 
         if ( watchedDirectory != watchedPaths_.end() ) {
+            const auto filename = fileInfo.fileName().toStdString();
 
-            auto watchedFile = std::find( watchedDirectory->files.begin(),
-                                          watchedDirectory->files.end(), filename );
+            std::vector<WatchedFile> files = watchedDirectory->files;
```

Comment:
Do we need a copy of files vector here?

Reply:
@variar, Yes, you noticed the problem correctly, it was more likely needed indicate type `auto&`

Reply:
@variar, fixed here: https://github.com/variar/klogg/commit/5d5153ad2c6c5af23e1b27fd179612978aef686b